### PR TITLE
Lisa report modal

### PR DESCRIPTION
**Pull request by 2016lisali**
1. add reportForm component
2. add a report Modal and add a report icon Button to the postCard to toggle the report Modal
3. add verifyEmail function and sendMessage Function in Contact page and add test cases for them
4. add tool-tips for icon buttons

### CODE REVIEW

**Comment by B08waffles** [comment]
This looks like really clean code, I'd love to see it merged however, how should we fix the maxed out 200 request per month email service? I wonder if there is a  different provider with a higher maximum requests per month or perhaps we we could loop through a list of keys/credentials in the case that one set of credentials has already been spent for the month to get these tests passed? I'm just throwing out some ideas, I  would love to hear what everyone else thinks.

**Review by B08waffles — APPROVED** [review]

**Comment by 2016lisali** [comment]
> This looks like really clean code, I'd love to see it merged however, how should we fix the maxed out 200 request per month email service? I wonder if there is a different provider with a higher maximum requests per month or perhaps we we could loop through a list of keys/credentials in the case that one set of credentials has already been spent for the month to get these tests passed? I'm just throwing out some ideas, I would love to hear what everyone else thinks.

I modified the test code and used mockedAxios for testing so that the tests will not send the request to API and consume the maximum requests.  
I've read this from an article:  'A common thing is for our application to fetch some data from the API. A problem with it is that might fail for various reasons, such as the API being down. We want our tests to be reliable and independent and to ensure that we can mock some of our modules.'. So mock Axios should be a good practice :)

**Review by QiujuYang — APPROVED** [review]

**Review by galarist — APPROVED** [review]
Awesome work!

File: src/pages/__tests__/Contact.test.tsx [hunk 1]
```diff
@@ -17,3 +31,102 @@ describe("Test Contact Page", () => {
         expect(wrapper.find("ContactForm").length).toBe(1);
     });
 });
+
+// test sendMessage function
+describe("Test sendMessage function", () => {
+    const setIsError = jest.fn();
```
Comment (Jet004) [inline src/pages/__tests__/Contact.test.tsx:37]: If you will reuse a mock then you should jest.resetAllMocks() before each test. Another option is to add to the jest configuration file to reset all mocks. This resets the mock to a clear state so tests can't interfere with each other.

File: src/data.tsx [hunk 2]
```diff
@@ -7,7 +7,14 @@ export const positionOptions = [
...
     "Software Engineer",
 ];
-
+export const violationOptions = [
+    "advertisement",
+    "Harassment or hate speech",
+    "Inappropriate listings message/category",
+    "Rude or vulgar",
+    "Spam or copyright issue",
+    "other",
```
Comment (Jet004) [inline src/data.tsx:16]: Suggestion: Capialise the O in other to be consistent with the other options

**Pull request merged**